### PR TITLE
Added number formatters

### PR DESCRIPTION
**Pull request by IvanSanchez**
I'm using this crate to display numbers in a few different displays (4- and 6-digit).

I've made some helper functions to calculate the byte arrays needed to represent a number, and I thought it'd be very nice to have those in the crate as well.

Docstrings and demo included.

### CODE REVIEW

**Review by JadKHaddad — CHANGES_REQUESTED** [review]
Thank you very much for your PR. Can you please run ``cargo fmt`` on the whole crate and then ``cargo test --features "async,blocking,demo,formatters"`` after taking a look at my suggestions?

File: README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ The following features are available:
 - `impl-debug`: implements `core::fmt::Debug` for structs and enums.
 - `impl-defmt-format`: implements `defmt::Format` for structs and enums.
 - `mappings`: enables the mappings module.
+- `formatters`: enables the formatters module.
```
Comment (JadKHaddad) [inline README.md]: Can you add this line to ``lib.rs`` as well?

File: tm1637/src/formatters.rs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+//! that can be sent to a TM1637 display.
+//!
+//! There are versions of these functions that are meant for 4-digit displays
+//! and for 6-digit displays. The 6-digit versions take into account that the
+//! order of the bytes does not drectly correlate with the order of the physical
+//! digits.
+//!
+//! All numbers are aligned to the right.
+//!
+//! #Example
```
Comment (JadKHaddad) [inline tm1637/src/formatters.rs]: Maybe like this?

```txt
//! ## Example
//!
//! ```rust,ignore
//! tm.write_bytes_raw(0, i16_to_4digits(1234));
//! ```
//!
```

File: tm1637/src/formatters.rs [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+//!
+//! ```
+//! tm.write_bytes_raw(0, i16_to_4digits(1234) );
+//! ```
+//!
+
+
+// Duplicated from mappings::DigitBits::from_digit
```
Comment (JadKHaddad) [inline tm1637/src/formatters.rs]: Maybe like this?

```txt
/// Convert a digit to byte.
#[cfg_attr(
    feature = "mappings",
    doc = "Duplicated from [`DigitBits::from_digit`](crate::mappings::DigitBits::from_digit)."
)]
```

File: tm1637/src/formatters.rs [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+		4 => 0b01100110,
+		5 => 0b01101101,
+		6 => 0b01111101,
+		7 => 0b00000111,
+		8 => 0b01111111,
+		9 => 0b01101111,
+		_ => 0b00111111,
+	}
+}
+
```
Comment (JadKHaddad) [inline tm1637/src/formatters.rs]: Can we document this function?

File: tm1637/src/formatters.rs [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+		5 => 0b01101101,
+		6 => 0b01101111,
+		7 => 0b00111000,
+		8 => 0b01111111,
+		9 => 0b01111101,
+		_ => 0b00111111,
+	}
+}
+
+/// Formats a `i16` clamped between -999 and 9999, for a 4-digit display
```
Comment (JadKHaddad) [inline tm1637/src/formatters.rs]: Using [i16], we can link to i16 directly from our crate ;)

File: tm1637/src/formatters.rs (hunk 3 above)
Comment (IvanSanchez) [reply tm1637/src/formatters.rs]: I'm wondering if I should instead call functionality from the `mappings` module, and make the `formatters` feature depend on the `mappings` feature. Do you have any thoughts about that?

File: tm1637/src/formatters.rs (hunk 4 above)
Comment (IvanSanchez) [reply tm1637/src/formatters.rs]: I'm wondering about adding a `UpsideDownDigits` struct to the mappings module instead. :thinking:

File: tm1637/src/formatters.rs (hunk 3 above)
Comment (JadKHaddad) [reply tm1637/src/formatters.rs]: Yes, I guess that makes sense. I thought it wasn't worth pulling in the whole ``mappings`` module just for one function. But now that you mention it, I guess it is a better idea to keep the mappings in the ``mappings`` module. Implementing other formatters like ``str_formatters`` would be cleaner this way.

File: tm1637/src/formatters.rs (hunk 4 above)
Comment (JadKHaddad) [reply tm1637/src/formatters.rs]: Yes 🤔

**Comment by IvanSanchez** [comment]
So, I've ran `cargo fmt`, I've tested every function with actual hardware, I've added the upside-down digits to `mappings`, I've changedthe feature dependencies, I've set up the `docs.rs` options so that docs are built for those features, I've made photographs of my test displays running the number formatters (I'd be nice to have those in the docs)... am I forgetting anything?

File: assets/4digits-rust.webp
Comment (JadKHaddad) [inline assets/4digits-rust.webp:1]: I love this one

**Review by JadKHaddad — APPROVED** [review]

**Comment by JadKHaddad** [comment]
@IvanSanchez Clippy 🫤

File: tm1637/src/formatters.rs [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+//! and for 6-digit displays. The 6-digit versions take into account that the
+//! order of the bytes does not drectly correlate with the order of the physical
+//! digits.
+//!
+//! All numbers are aligned to the right.
+//!
+//! # Example
+//!
+//! ```rust,ignore
+//! tm.write_bytes_raw(0, i16_to_4digits(1234) );
```
Comment (JadKHaddad) [inline tm1637/src/formatters.rs]: I can't unsee the whitespace in ``tm.write_bytes_raw(0, i16_to_4digits(1234) );`` can you remove it please 😁

**Review by JadKHaddad — CHANGES_REQUESTED** [review]

**Review by JadKHaddad — APPROVED** [review]

**Pull request merged**